src/network/networkd-dhcp-common.c

```diff
@@ -343,6 +346,8 @@ int config_parse_dhcp_send_option(
                 return 0;
         }
 
+        dhcpv6 = !(STRPTR_IN_SET(section, "DHCP", "DHCPv4"));
```

Comment:
instead of matching the section like this: use the `ltype` paremeter for this. ltype is a general purpose int parameter that can be used for things like this. just set it to the address family (AF_INET, AF_INET6, …) int the config parser table. That should be much sexier.